src/db.rs

```diff
@@ -71,6 +71,59 @@ impl DB {
         Ok(())
     }
 
+    pub fn migrate<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        if !self.dirs.is_empty() {
+            return Err(anyhow!(
+                "To prevent conflicts, you can only migrate from z with an empty \
+                 zoxide database!"
+            ));
+        }
+
+        let zdata = File::open(path)?;
+        let reader = BufReader::new(zdata);
```

Comment:
I'm normally not a fan of short variables, but naming readers `rd` for brevity seems to be standard in Rust code. Could you rename this to `rd`?

Reply:
Honestly, I'm not a fan of renaming it to `rd` -- solely because it goes against what you asked me above when opening the file ("a more descriptive name"). I can't claim to have read a lot of Rust codebases, but I haven't ever seen a reader named `rd`.

Unless you very strongly insist, I think I'll refrain from making this change.

Reply:
I did notice the irony, but I've seen something `fn foo<R: Read>(rd: R)` in a lot of places.

However, I just checked https://github.com/rust-lang/rust, and it seems they're using `reader` for their `BufReader`s. So, you win!